lib/games/games_module.rb

```diff
@@ -17,19 +16,44 @@ def add_game
     print 'Please state how many years ago it was published: '
     published_date = gets.chomp
     new_game = Game.new(title, multiplayer, last_played_at, published_date)
-    @game_list.push(new_game)
+    @game_list.push(title: new_game.title, multiplayer: new_game.multiplayer, last_played_at: new_game.last_played_at, published_date: new_game.published_date)
     puts 'Game created succesfully'
     puts ''
   end
 
   def list_all_games
-    @game_list.each do |game|
-      puts "Title: \"#{game.title}\", Multiplayer: #{game.multiplayer}, Last Time Played:#{game.last_played_at} years ago, Published #{game.published_date} years ago"
+    @game_list = JSON.parse(File.read('./lib/games/json_data/games.json'))
+    if @game_list.empty?
+      puts 'There are no games in the list yet, please add one first'
+    else
+      @game_list.each do |game|
+        puts "Title: #{game['title']}, Multiplayer: #{game['multiplayer']}, Last Time Played:#{game['last_played_at']} years ago, Published #{game['published_date']} years ago"
+      end
     end
   end
 
   def list_all_authors
-    puts 'There are no authors in the list yet, please add one first'
+    @authors_list = JSON.parse(File.read('./lib/games/json_data/authors.json'))
+    if @authors_list.empty?
+      puts 'There are no games in the list yet, please add one first'
+    else
+      @authors_list.each do |author|
+        puts "Title: #{author['first_name']}, Multiplayer: #{author['last_name']}"
+      end
+    end
+  end
+
+  def read_games_data
+    if File.exist?('./lib/games/json_data/games.json')
+      File.open('./lib/games/json_data/games.json', 'a') { |f| f.write(JSON.generate([])) }
+    elsif File.exist?('./lib/games/json_data/authors.json')
+      File.open('./lib/games/json_data/authors.json', 'a') { |f| f.write(JSON.generate([])) }
+    end
+  end
+
+  def save_games_data
+    File.open('./lib/games/json_data/games.json', 'w+') { |f| f.write(JSON.generate(game_list)) }
+    File.open('./lib/games/json_data/authors.json', 'w+') { |f| f.write(JSON.generate(@authors_list)) }
   end
```

Comment:
We have the same problem here when this file is not present in the document this would throw an exception. I suggest writing a file if the file doesn't exist, and since we're always saving when exiting the app you don't need to worry about appending to a file you can overwrite it.

Error: Exiting when the file doesn't exist:

![image](https://user-images.githubusercontent.com/85680436/144168438-df06ccb4-16b4-4faf-9059-4688a0302a5c.png)